src/transmute.rs

```diff
@@ -43,3 +58,46 @@ impl LateLintPass for UselessTransmute {
         }
     }
 }
+
+pub struct CrosspointerTransmute;
+
+impl LintPass for CrosspointerTransmute {
+    fn get_lints(&self) -> LintArray {
+        lint_array!(CROSSPOINTER_TRANSMUTE)
+    }
+}
+
+fn is_ptr_to(from: &TyS, to: &TyS) -> bool {
+    if let TyRawPtr(from_ptr) = from.sty {
+        from_ptr.ty == to
+    } else {
+        false
+    }
+}
+
+impl LateLintPass for CrosspointerTransmute {
+    fn check_expr(&mut self, cx: &LateContext, e: &Expr) {
+        if let ExprCall(ref path_expr, ref args) = e.node {
+            if let ExprPath(None, _) = path_expr.node {
+                let def_id = cx.tcx.def_map.borrow()[&path_expr.id].def_id();
+
+                if utils::match_def_path(cx, def_id, &["core", "intrinsics", "transmute"]) {
```

Comment:
You should put that path with the [big ugly list of other path](https://github.com/Manishearth/rust-clippy/blob/7e65493599d631f5be0fbc882974153bf85a5ad6/src/utils/mod.rs#L23-L55).